main.go

```diff
@@ -92,16 +93,23 @@ func main() {
 	osClient := osclientset.NewForConfigOrDie(rest.AddUserAgent(config, controllers.ComponentName))
 	kubeClient := kubernetes.NewForConfigOrDie(rest.AddUserAgent(config, controllers.ComponentName))
 
+	enabledFeatures, err := controllers.EnabledFeatures(context.Background(), osClient)
+	if err != nil {
+		klog.ErrorS(err, "unable to get enabled features")
+		os.Exit(1)
+	}
+
 	enableWebhook := provisioning.WebhookDependenciesReady(osClient)
 
 	if err = (&controllers.ProvisioningReconciler{
-		Client:         mgr.GetClient(),
-		Scheme:         mgr.GetScheme(),
-		OSClient:       osClient,
-		KubeClient:     kubeClient,
-		ReleaseVersion: releaseVersion,
-		ImagesFilename: imagesJSONFilename,
-		WebHookEnabled: enableWebhook,
+		Client:          mgr.GetClient(),
+		Scheme:          mgr.GetScheme(),
+		OSClient:        osClient,
+		KubeClient:      kubeClient,
+		ReleaseVersion:  releaseVersion,
+		ImagesFilename:  imagesJSONFilename,
+		WebHookEnabled:  enableWebhook,
+		EnabledFeatures: enabledFeatures,
```

Comment:
We can determine enabledFeatures while calling IsEnabled() in SetupWithManager().